internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -381,6 +381,53 @@ func isUserManagedNetwork(clusterInstall *hiveext.AgentClusterInstall) bool {
 	return false
 }
 
+func hyperthreadingInSpec(clusterInstall *hiveext.AgentClusterInstall) bool {
+	//see https://docs.openshift.com/container-platform/4.7/installing/installing_platform_agnostic/installing-platform-agnostic.html#installation-bare-metal-config-yaml_installing-platform-agnostic
+	return len(clusterInstall.Spec.Compute) > 0 || clusterInstall.Spec.ControlPlane != nil
+}
+
+func setHyperthreading(clusterInstall *hiveext.AgentClusterInstall, hyperthreadingMode **string) {
+	const (
+		None    = 0
+		Workers = 1
+		Masters = 2
+		All     = 3
+	)
+	var config uint = 0
```

Comment:
you can use just config variable to know if the update happened or not

Reply:
Yes. I already got rid of the update check. The external update check is needed so we can share code between UpdateCluster and RegisterCluster

Reply:
See the new logic with the default behavior in place